src/nodes/hypertable_insert.c

```diff
@@ -469,6 +478,32 @@ hypertable_insert_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *be
 	 * targetlist is set, we go back and fix up the CustomScan's targetlist.
 	 */
 	cscan->scan.plan.targetlist = copyObject(root->processed_tlist);
+
+	/*
+	 * For DELETE processed_tlist will have ROWID_VAR. We need to remove
+	 * those because set_customscan_references will bail if it sees
+	 * ROWID_VAR entries in the targetlist.
+	 */
+	if (mt->operation == CMD_DELETE)
+	{
+#if PG14_GE
+		ListCell *lc;
+		foreach (lc, cscan->scan.plan.targetlist)
+		{
+			TargetEntry *tle = lfirst_node(TargetEntry, lc);
+
+			if (IsA(tle->expr, Var) && castNode(Var, tle->expr)->varno == ROWID_VAR)
+			{
+				Var *var = castNode(Var, tle->expr);
+				tle->expr = (Expr *) makeNullConst(var->vartype, var->vartypmod, var->varcollid);
+			}
+		}
+#else
+		/* We only route DELETEs through our CustomNode for PG 14+ because
+		 * the codepath for earlier versions is different. */
+		Assert(false);
+#endif
+	}
```

Comment:
```suggestion
#if PG14_GE
	/*
	 * For DELETE processed_tlist will have ROWID_VAR. We need to remove
	 * those because set_customscan_references will bail if it sees
	 * ROWID_VAR entries in the targetlist.
	 */
	if (mt->operation == CMD_DELETE)
	{
		ListCell *lc;
		foreach (lc, cscan->scan.plan.targetlist)
		{
			TargetEntry *tle = lfirst_node(TargetEntry, lc);

			if (IsA(tle->expr, Var) && castNode(Var, tle->expr)->varno == ROWID_VAR)
			{
				Var *var = castNode(Var, tle->expr);
				tle->expr = (Expr *) makeNullConst(var->vartype, var->vartypmod, var->varcollid);
			}
		}
	}
#endif
```

Isn't this way simpler?

Reply:
i wanted to catch any non-pg14 deletes ending up here